src/regex.cr

```diff
@@ -511,22 +573,52 @@ class Regex
   # # `$~` is not set even if last match succeeds.
   # $~ # raises Exception
   # ```
-  def matches?(str, pos = 0, options = Regex::Options::None) : Bool
+  def matches?(str, pos = 0, options : Regex::MatchOptions = Regex::MatchOptions::None) : Bool
```

Comment:
Do you think the `str` and `pos` parameters should be restricted in the new overloads as well?

Also I believe the default argument can simply be `:none` as it will be autocasted

Reply:
Sounds like a good idea (both).